azure/services/scalesets/scalesets.go

```diff
@@ -564,11 +564,14 @@ func (s *Service) buildVMSSFromSpec(ctx context.Context, vmssSpec azure.ScaleSet
 			nicConfig := compute.VirtualMachineScaleSetNetworkConfiguration{}
 			nicConfig.VirtualMachineScaleSetNetworkConfigurationProperties = &compute.VirtualMachineScaleSetNetworkConfigurationProperties{}
 			nicConfig.Name = pointer.String(vmssSpec.Name + "-" + strconv.Itoa(i))
-			if pointer.BoolDeref(n.AcceleratedNetworking, false) {
-				nicConfig.VirtualMachineScaleSetNetworkConfigurationProperties.EnableAcceleratedNetworking = pointer.Bool(true)
+			nicConfig.EnableIPForwarding = pointer.Bool(true)
+
+			if n.AcceleratedNetworking == nil {
+				nicConfig.VirtualMachineScaleSetNetworkConfigurationProperties.EnableAcceleratedNetworking = vmssSpec.AcceleratedNetworking
```

Comment:
I'm not sure we should be defaulting the VMMS spec field in the controller since it's deprecated, and we have webhook logic in place to migrate the value of the old field to the new one https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/exp/api/v1beta1/azuremachinepool_default.go#L140 

maybe we're missing defaulting logic in the webhook?

Reply:
defaulting logic might help indeed but maybe a task for the refactoring?

setting it to `vmssSpec.AcceleratedNetworking` because that's being set anyway based on the SKU here:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3243/files#diff-a6779b0b37d0dcdbe7dcd6f00150a22e12ac067a96269038452d27ba39349a90L449